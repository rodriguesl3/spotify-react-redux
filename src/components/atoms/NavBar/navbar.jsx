/* eslint-disable react/prop-types */

import React from 'react';
import { Link } from 'react-router-dom';
import { MdPerson, MdSearch } from 'react-icons/md/index';
import logo from '../../../logo.svg';
import './navbar.scss';
import Search from '../Search/Search';

const NavBar = ({
  brandName,
  onShowSideBar,
  onIsSearching,
  isSearching,
  onSearchTracksArtists,
  searchResult
}) => {
  const showSideBarHandle = () => { onShowSideBar(true); };
  const showSearchArtist = () => {
    onIsSearching(!isSearching);
  };

  return (
    <div>
      <nav className="navbar navbar-dark bg-dark">
        <Link to="/login" className="linkContainer">
          <div className="link col-6">
            {brandName}
          </div>
          <div className="link col-6">
            <img src={logo} alt="Brand Icon" className="App-logo" />
          </div>
        </Link>
        <span onClick={showSideBarHandle} className="profileIcon" role="presentation">
          <MdPerson size={30} />
        </span>
        <span onClick={showSearchArtist} >
          <MdSearch />
        </span>
        {isSearching
          && (
            <Search
              onIsSearching={onIsSearching}
              isSearching={isSearching}
              onSearchTracksArtists={onSearchTracksArtists}
              searchResult={searchResult}
            />
          )
        }
      </nav>
    </div>
  );
};

export default NavBar;
