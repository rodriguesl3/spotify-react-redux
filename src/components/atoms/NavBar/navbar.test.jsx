/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import NavBar from './navbar';


test('SideBar snaphot test', () => {
  const component = renderer.create(
    <MemoryRouter>
      <NavBar brandName="Spotify React" />
    </MemoryRouter>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Search snaphot test', () => {
  const component = renderer.create(
    <MemoryRouter>
      <NavBar brandName="Spotify React" isSearching />
    </MemoryRouter>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
