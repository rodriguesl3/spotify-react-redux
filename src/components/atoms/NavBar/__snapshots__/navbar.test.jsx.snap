// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search snaphot test 1`] = `
<div>
  <nav
    className="navbar navbar-dark bg-dark"
  >
    <Link
      className="linkContainer"
      replace={false}
      to="/login"
    >
      <div
        className="link col-6"
      >
        Spotify React
      </div>
      <div
        className="link col-6"
      >
        <img
          alt="Brand Icon"
          className="App-logo"
          src="logo.svg"
        />
      </div>
    </Link>
    <span
      className="profileIcon"
      onClick={[Function]}
      role="presentation"
    >
      <MdPerson
        size={30}
      />
    </span>
    <span
      className="searchArtist"
      onClick={[Function]}
    >
      <MdSearch />
    </span>
    <Search
      isSearching={true}
      onIsSearching={[MockFunction]}
      onSearchTracksArtists={[MockFunction]}
      searchResult={Object {}}
    />
  </nav>
</div>
`;

exports[`shallow navbar 1`] = `
<div>
  <nav
    className="navbar navbar-dark bg-dark"
  >
    <Link
      className="linkContainer"
      replace={false}
      to="/login"
    >
      <div
        className="link col-6"
      >
        Spotify React
      </div>
      <div
        className="link col-6"
      >
        <img
          alt="Brand Icon"
          className="App-logo"
          src="logo.svg"
        />
      </div>
    </Link>
    <span
      className="profileIcon"
      onClick={[Function]}
      role="presentation"
    >
      <MdPerson
        size={30}
      />
    </span>
    <span
      className="searchArtist"
      onClick={[Function]}
    >
      <MdSearch />
    </span>
    <Search
      isSearching={true}
      onIsSearching={[MockFunction]}
      onSearchTracksArtists={[MockFunction]}
      searchResult={Object {}}
    />
  </nav>
</div>
`;
